src/device-registry/utils/create-event.js

```diff
@@ -1675,12 +1675,13 @@ const createEvent = {
 
     if (!responseFromTransformMeasurements.success) {
       logger.error(
-        `internal server error -- unable to transform measurements -- ${responseFromTransformMeasurements.message}`
+        `internal server error -- unable to transform measurements -- ${responseFromTransformMeasurements.message}, ${JSON.stringify(measurements)}`
       );
     }
 
-    for (const measurement of responseFromTransformMeasurements.data) {
+    for (const measurementdata of Object.keys(responseFromTransformMeasurements.data)) {
```

Comment:
Hi @BenjaminSsempala , could we first remove this modification to the function as we continue to study the issue by observing the enhanced logging statement?

Reply:
Alright @Baalmart   I've made these changes.